aioxmpp/utils.py

```diff
@@ -56,15 +63,79 @@
     "namespaces",
 ]
 
-namespaces = types.SimpleNamespace()
+
+class Namespaces:
+    """
+    Manage short-hands for XML namespaces.
+
+    Instances of this class may be used to assign mnemonic short-hands
+    to XML namespaces, for example::
+
+        namespaces = Namespaces()
+        namespaces.foo = "urn:example:foo"
+        namespaces.bar = "urn:example:bar"
+
+    The class ensures that
+
+    1. only one short-hand is bound to each namespace, so continuing the
+       example, the following raises :class:`ValueError`::
+
+           namespace.example_foo = "urn:example:foo"
+
+    2. no short-hand is redefined to point to a different namespace,
+       continuing the example, the following raises
+       :class:`ValueError`::
+
+           namespaces.foo = "urn:example:foo:2"
+
+    3. deleting a short-hand is prohibited, the following raises
+       :class:`TypeError`::
```

Comment:
Maybe AttributeError would be more appropriate? That’s what properties do, too:

```python
>>> class Foo:
...     @property
...     def bar(self):
...         return "x"
...
>>> f = Foo()
>>> del f.bar
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't delete attribute
```

Reply:
Probably, I checked a built-in (which raises TypeError), but the analogy to properties is more appropriate.